File: src/main/java/org/goafabric/personservice/controller/dto/Address.java
package org.goafabric.personservice.controller.dto;

import jakarta.annotation.Nullable;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;

public record Address (
    @Nullable String id,
    @Nullable Long version,
    @NotNull @Size(min = 3, max = 255) @Pattern(regexp = "[a-zA-Z0-9.\\s]*") String street,
    @NotNull @Size(min = 3, max = 255) String city) {
}

File: src/main/kotlin/org/goafabric/personservice/controller/dto/Address.kt
package org.goafabric.personservice.controller.dto

import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size

data class Address (
    val id: String? = null,
    val version: String? = null,
    @field:Size(min = 3, max = 255) @field:Pattern(regexp = "[a-zA-Z0-9.\\s]*") val street: String,
    @field:Size(min = 3, max = 255) val city: String
)